File: Eclipse/w7cpframe/src/clsPfSection.java
//
// Copyright (c)2016 S C Harrison
// Refer to License.txt for terms and conditions of use.
//
import java.io.*;
import java.util.regex.Matcher;
import java.util.regex.Pattern;



    class clsPfSection
    {

        //Define Class

        public int key = 0;

        public double ax = 0;           //.. member's cross sectional area ..
        public double iz = 0;           //.. member's second moment of area ..

        //Dependent on Material Properties
        public double t_len = 0;        //.. TOTAL length of this section ..
        public double t_mass = 0;       //.. TOTAL mass of this section ..

        public int mat = 0;          //.. material of section ..

        public String Descr = "";       //.. section description String ..



        public void initialise()
        {
            key = 0;
            ax = 0;
            iz = 0;
            mat = 0;
            Descr = "<unknown>";

            t_len = 0;
            t_mass = 0;

        }

        public void setValues(int sectionKey, double SectionArea, double SecondMomentArea, int materialKey, String Description)
        {
            key = sectionKey;
            ax = SectionArea;
            iz = SecondMomentArea;
            mat = materialKey;
            Descr = Description;
        }

        public String sprint()
        {
            String s;

            s = "";
            s = s + String.format("%8d",key);
            s = s + String.format("%15.4f",ax);
            s = s + String.format("%15.4f",iz);
            s = s + String.format("%6d",mat);
            s = s + String.format("%28s",Descr);

            return s;

        }


        public void cprint()
        {
            System.out.println(sprint());
        }

        public void fprint(BufferedWriter fp)
        {
            try {
				fp.write(sprint());
			} catch (IOException e) {
				// TODO Auto-generated catch block
				e.printStackTrace();
			}
        }


        public void fgetData(BufferedReader fp)
        {
            String s="";


            System.out.println("fgetData ...");

            try {
				s = fp.readLine();
			} catch (IOException e) {
				// TODO Auto-generated catch block
				e.printStackTrace();
			}
            //WScript.Echo( s);
            sgetData(s);

            System.out.println("... fgetData");
        }


        public void sgetData(String s)
        {
            String [] dataflds = new String[10]; //(0 To 9);
            int i,n;
            String s1;

            System.out.println("sgetData ...");
            System.out.println(s);

            //Trim Trailing Spaces from input String
            String regExpTrimPattern = "^\\s+|\\s+$"; //trim trailing spaces          
            Pattern regExpPatternObj = Pattern.compile(regExpTrimPattern);
            Matcher regExpMatches = regExpPatternObj.matcher(s);
            String regResult = regExpMatches.replaceAll("");
            
            //Get Data fields from input String
            String regExpNumberPattern = "-?\\d+(?:[,.]\\d+)?";
            regExpPatternObj = Pattern.compile(regExpNumberPattern);
            regExpMatches = regExpPatternObj.matcher(regResult);
            i=0;
            while (regExpMatches.find()){
            	dataflds[i] = regExpMatches.group().trim();
            	i = i+1;
            }

            s1 = "";
            n = dataflds.length;
            for (i = 0; i < n; i++)
            {
                System.out.println(i + "<" + dataflds[i] + ">");
                if (dataflds[i] != "")
                {
                    if (s1 == "")
                    {
                        s1 = dataflds[i];
                    }
                    else
                    {
                        s1 = s1 + "," + dataflds[i];
                    }
                }
            }
            System.out.println(s1);
            System.out.println();

            n = dataflds.length;
            for (i = 0; i < n; i++)
            {
                System.out.println(i + "<" + dataflds[i] + ">");
            }

            key = Integer.parseInt(dataflds[0]);
            ax = Double.parseDouble(dataflds[1]);
            iz = Double.parseDouble(dataflds[2]);
            mat = Integer.parseInt(dataflds[3]);
            Descr = dataflds[4];

            //Zero Variables
            t_len = 0;
            t_mass = 0;

            System.out.println("... sgetData");
        }





    } //class

File: VisualStudio/VBasic/w7cpframe/clsPfSection.vb
'
' Copyright (c)2016 S C Harrison
' Refer to License.txt for terms and conditions of use.
'
Option Explicit On

Public Class clsPfSection

#Const isUsingExcel = 0

    Public key As Integer

    Public ax As Double          '.. member's cross sectional area ..
    Public iz As Double         '.. member's second moment of area ..

    'Dependent on Material Properties
    Public t_len As Double       '.. TOTAL length of this section ..
    Public t_mass As Double      '.. TOTAL mass of this section ..
    Public mat As Byte           '.. material of section ..

    Public Descr As String       '.. section description string ..

    Sub initialise()
        ax = 0
        iz = 0
        t_len = 0
        t_mass = 0
        mat = 0
        Descr = "<unknown>"
    End Sub

    Sub setValues(ByVal sectionKey As Integer, ByVal SectionArea As Double, ByVal SecondMomentArea As Double, ByVal materialKey As Integer, ByVal Description As String)
        key = sectionKey
        ax = SectionArea
        iz = SecondMomentArea
        mat = materialKey
        Descr = Description
    End Sub

    Function sprint() As String
        sprint = StrLPad(Format(key, "##0"), 8) _
                 & StrLPad(Format(ax, "0.0000E+00"), 15) _
                 & StrLPad(Format(iz, "0.0000E+00"), 15) _
                 & StrLPad(Format(mat, "##0"), 6) _
                 & StrLPad(Descr, 28)

    End Function


    Sub cprint()
        Console.WriteLine(sprint)
    End Sub

    Sub fprint(ByVal fp As Integer)
        Print(fp, sprint)
    End Sub


#If isUsingExcel = 1 Then

    Sub wbkPrint(ByVal dataRec As Range, ByVal RecNum As Byte)
        With dataRec
            .Offset(RecNum, 0).Value = key
            .Offset(RecNum, 1).Value = ax
            .Offset(RecNum, 2).Value = iz
            .Offset(RecNum, 3).Value = mat
            .Offset(RecNum, 4).Value = Descr

            '      .Offset(RecNum, 3).value = t_len
            '      .Offset(RecNum, 4).value = t_mass

        End With
    End Sub

    Sub wbkRead(ByVal dataRec As Range, ByVal RecNum As Byte)
        With dataRec
            key = .Offset(RecNum, 0).Value
            ax = .Offset(RecNum, 1).Value
            iz = .Offset(RecNum, 2).Value
            mat = .Offset(RecNum, 3).Value
            Descr = .Offset(RecNum, 4).Value
            '      t_len = .Offset(RecNum, 3).value
            '      t_mass = .Offset(RecNum, 4).value

        End With
    End Sub

#End If

End Class
